### PR TITLE
Display substation properties

### PR DESCRIPTION
**Pull request by anistouri**

### CODE REVIEW

File: src/components/spreadsheet/utils/customTooltipKeyValue.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { Grid, Typography, Tooltip } from '@mui/material';
+import makeStyles from '@mui/styles/makeStyles';
+import { ITooltipParams } from 'ag-grid-community';
+import { FunctionComponent } from 'react';
+
+const useStyles = makeStyles(() => ({
```
Comment (flomillot) [inline src/components/spreadsheet/utils/customTooltipKeyValue.tsx:6]: No more makestyles, use sx instead

File: src/components/spreadsheet/utils/customTooltipKeyValue.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    let keys: string[] = [];
+    if (properties) {
+        keys = Object.keys(properties);
+    }
+    return keys.length > 2 ? (
+        <Tooltip title={value.title}>
+            <Grid container item xs={6} className={classes.container}>
+                {keys.map((key) => {
+                    return (
+                        <Grid container spacing={1} className={classes.item}>
```
Comment (flomillot) [inline src/components/spreadsheet/utils/customTooltipKeyValue.tsx:33]: Browser don't complain that you don't use key after your map ?

File: src/components/spreadsheet/utils/customTooltipKeyValue.tsx (hunk 1 above)
Comment (anistouri) [reply src/components/spreadsheet/utils/customTooltipKeyValue.tsx]: Done

File: src/components/spreadsheet/utils/customTooltipKeyValue.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+const CustomTooltipKeyValue: FunctionComponent<ITooltipParams> = (props) => {
+    const classes = useStyles();
+    const value = props?.value;
+    let properties = value?.properties;
+    let keys: string[] = [];
+    if (properties) {
+        keys = Object.keys(properties);
+    }
+    return keys.length > 2 ? (
```
[the remark this answers is not in the document]
Comment (anistouri) [reply src/components/spreadsheet/utils/customTooltipKeyValue.tsx:12]: In TypeScript, when the component type is defined as FunctionComponent, you are committing to return either a React element (ReactElement) or null. Returning a boolean violates this type constraint

File: src/components/spreadsheet/utils/customTooltipKeyValue.tsx (hunk 2 above)
Comment (anistouri) [reply src/components/spreadsheet/utils/customTooltipKeyValue.tsx]: Indeed, I have added it. Thanks

File: src/components/spreadsheet/utils/customTooltipKeyValue.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (flomillot) [reply src/components/spreadsheet/utils/customTooltipKeyValue.tsx:12]: Ah OK thanks

File: src/components/spreadsheet/utils/sensi-properties.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { ICellRendererParams } from 'ag-grid-community';
+
+interface LinesType {
+    [key: string]: string;
+}
+export class SensiProperties {
```
Comment (flomillot) [inline src/components/spreadsheet/utils/sensi-properties.ts:6]: Did you get part or whole of this code from the AGGrid documentation ? If yes, can you add the link to the page ?
It's usefull if one day the code brake.

**Review by trikimahdi — CHANGES_REQUESTED** [review]

**Review by flomillot — APPROVED** [review]

**Review by trikimahdi — APPROVED** [review]
Code ok
test ok

**Pull request merged**